Dockerfile.app

```diff
@@ -3,26 +3,25 @@ ARG NPM_VERSION="5.5.0"
 FROM registry.devshift.net/fabric8-hdd/nodejs:${NODE_VERSION}_npm_${NPM_VERSION}
 LABEL maintainer="Anmol Babu <anmolbudugutta@gmail.com>"
 
-# Setup App Dir
-RUN mkdir -p /usr/src/app
-WORKDIR /usr/src/app
-
 # Install git for npm to be able to pull dependencies from git
 RUN yum -y install git
 
 # Install app dependencies
 ADD package.json ./package.json
 ADD package-lock.json ./package-lock.json
-
-# Install app
 RUN npm install --production
 
+# Setup App Dir
+RUN mkdir -p /usr/src/app && cp -a ./node_modules /usr/src/app/
```

Comment:
Why not use mv instead of cp?

Also, with cp, it would be good to use the -R ( --recursive ) flag to ensure the sub-directory structure.

Reply:
mv is a good idea..

man extract of cp : 

NAME
       cp - copy files and directories

SYNOPSIS
       cp [OPTION]... [-T] SOURCE DEST
       cp [OPTION]... SOURCE... DIRECTORY
       cp [OPTION]... -t DIRECTORY SOURCE...

DESCRIPTION
       Copy SOURCE to DEST, or multiple SOURCE(s) to DIRECTORY.

       Mandatory arguments to long options are mandatory for short options too.

       -a, --archive
              same as -dR --preserve=all

so cp -a is better than cp -R